generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/generateUnion.kt

```diff
@@ -29,25 +30,49 @@ import graphql.schema.GraphQLUnionType
 import kotlin.reflect.KClass
 import kotlin.reflect.full.createType
 
-internal fun generateUnion(generator: SchemaGenerator, kClass: KClass<*>): GraphQLUnionType {
+internal fun generateUnion(generator: SchemaGenerator, kClass: KClass<*>, unionAnnotation: GraphQLUnion? = null): GraphQLUnionType {
+    return if (unionAnnotation != null) {
+        generateUnionFromAnnotation(generator, unionAnnotation)
+    } else {
+        generateUnionFromKClass(generator, kClass)
+    }
+}
+
+private fun generateUnionFromAnnotation(generator: SchemaGenerator, unionAnnotation: GraphQLUnion): GraphQLUnionType {
     val builder = GraphQLUnionType.newUnionType()
-    builder.name(kClass.getSimpleName())
+    builder.name(unionAnnotation.name)
+    builder.description(unionAnnotation.description)
+
+    val possibleTypes = unionAnnotation.possibleTypes.toList()
+
+    return createUnion(generator, builder, possibleTypes, unionAnnotation.name)
```

Comment:
with this new annotation it looks like it is not possible to specify union directives?

Reply:
Yea, not in this current state. I wonder though if there is some other work around we could do. Would this be better?

```kotlin
@GrapQLUnion(possibleTypes = [SharedModel::class, ServiceModel::class])
interface CustomUnion

@Suppress("CAST_NEVER_SUCCEEDS")
fun model(): CustomUnion = SharedModel("foo") as CustomUnion
```

Reply:
since `SharedModel` does not implement this marker interface this won't work -> `Any` workaround was added specifically for that